dotCMS/src/main/java/com/dotmarketing/business/cache/provider/CacheSizingUtil.java

```diff
@@ -0,0 +1,285 @@
+            Random r = new Random();
+            Object object = cacheMap.get(keysAsArray.get(r.nextInt(keysAsArray.size())));
+            if (object instanceof Optional && ((Optional) object).isPresent()) {
+                object = ((Optional) object).get();
+            }
+            if (object instanceof Optional && !((Optional) object).isPresent()) {
+                return 0;
+            }
+            totalSize += sizeOf(object);
+        }
+        return totalSize / numberToSample;
+
+    }
+
+    /**
+     * Tries to use the Unsafe class to calculate an objects size. If this does not work, it will fall
+     * back on the serialzied size (which less than accurate).
+     * 
+     * @param cacheMap
+     * @return
+     */
+    public long sizeOf(Object object) {
+        long retainedSize = 0;
+        try {
+            retainedSize = retainedSize(object);
+        } catch (Throwable t) {
+            Logger.infoEvery(this.getClass(),
+                            "Unable to use the Unsafe.class to size cache objects, falling back to serialization:"
+                                            + t.getMessage(),
+                            60000);
+        }
+
+        return retainedSize > 0 ? retainedSize : sizeOfSerialized(object);
+
+    }
+
+
+    final ByteArrayOutputStream byteOutputStream = new ByteArrayOutputStream();
+
+    /**
+     * Serializes an object to attempt to find an object's size
+     * 
+     * @param object
+     * @return
+     */
+    long sizeOfSerialized(Object object) {
+
+        if (object == null) {
+            return 0;
+        }
+
+        if (!(object instanceof Serializable)) {
+            Logger.warn(this.getClass(), object.getClass() + " not serializable: " + object);
+            return -1;
+        }
+
+        if (object instanceof String) {
+            return ((String) object).length() * 8;
+
+        }
+
+        byteOutputStream.reset();
+        try (ObjectOutputStream objectOutputStream = new ObjectOutputStream(byteOutputStream)) {
+            objectOutputStream.writeObject(object);
+            objectOutputStream.close();
+
+            return byteOutputStream.toByteArray().length;
+        } catch (Exception e) {
+            Logger.warnAndDebug(this.getClass(), "failed writing:" + object.getClass() + " : " + e.getMessage(), e);
+            return 0;
+        }
+
+    }
+
+    /**
+     * returns the size of an objects class, based on the classes properties
+     * 
+     * @param object
+     * @return
+     */
+    long sizeOfClass(Object object) {
+
+        return ClassLayout.parseClass(object.getClass()).instanceSize();
+    }
+
+    public static boolean useCompressedOops = true;
+
+    public static int retainedSize(Object obj) {
+        return retainedSize(obj, new HashMap<>(), 0);
+    }
+
+
+
+
+    @SuppressWarnings("restriction")
+    private static int retainedSize(Object obj, HashMap<Object, Object> calculated, final int depth) {
```

Comment:
Codacy found an issue: [Avoid using implementation types like 'HashMap'; use the interface instead](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6315945)